lib/block-supports/colors.php

```diff
@@ -5,6 +5,47 @@
  * @package gutenberg
  */
 
+/**
+ * Registers the style and colors block attributes for block types that support it.
+ *
+ * @param  array $block_type Block Type.
+ */
+function gutenberg_register_colors_support( $block_type ) {
+	$color_support                 = gutenberg_experimental_get( $block_type->supports, array( '__experimentalColor' ), false );
+	$has_text_colors_support       = is_array( $color_support ) || $color_support;
+	$has_background_colors_support = $has_text_colors_support;
+	$has_gradients_support         = $has_text_colors_support && gutenberg_experimental_get( $color_support, array( 'gradients' ), false );
+
+	if ( ! $block_type->attributes ) {
+		$block_type->attributes = array();
+	}
+
+	if ( $has_text_colors_support && ! array_key_exists( 'style', $block_type->attributes ) ) {
```

Comment:
This may be more readable if we use `$color_support` instead of `$has_text_colors_support` (or a comment), since the style attribute needs to be initialized for custom values of all the different items that can have color support.  

At first glance reading it I thought "why are we only defining the style object for text colors?", but then realized that none of the others can be supported without text color support anyways.  So it still looks correct from a functionality standpoint!